invenio_rdm_records/collections/service.py

```diff
@@ -121,6 +127,31 @@ def add(self, identity, collection, slug, title, query, uow=None, **kwargs):
             identity, new_collection, self.collection_schema, self.links_item_tpl
         )
 
+    @unit_of_work()
+    def update(self, identity, collection_or_id, data=None, uow=None):
+        """Update a collection."""
+        if isinstance(collection_or_id, int):
+            collection = self.collection_cls.resolve(id_=collection_or_id)
+        else:
+            collection = collection_or_id
+        self.require_permission(
+            identity, "update", community_id=collection.community.id
+        )
+
+        data = data or {}
+
+        valid_data, errors = self.collection_schema.load(
+            data, context={"identity": identity}, raise_errors=True
+        )
+
+        res = collection.update(**valid_data)
```

Comment:
The schema definition is already "protecting" what can be updated. Do we need the extra check inside the method to allow only certain attr to be updated?

Reply:
I prefer the model to be protected because it “knows” which fields are safe to modify. For instance, depth is calculated based on the path during creation, and the path isn’t updated since we don’t rearrange the tree. Currently, these fields aren’t in the schema, but there’s a risk that they could be added in the future, inadvertently introducing bugs. To me, I am not a big fan of relying on Marshmallow because the schema feels “far” from the model and it’s also used in other ways beyond validating what can and cannot be updated.

But I think it's more of a personal fear so I will remove it.